code/modules/mod/mod.dm

```diff
@@ -0,0 +1,549 @@
+		QDEL_NULL(helmet)
+	if(chestplate)
+		chestplate.mod = null
+		QDEL_NULL(chestplate)
+	if(gauntlets)
+		gauntlets.mod = null
+		QDEL_NULL(gauntlets)
+	if(boots)
+		boots.mod = null
+		QDEL_NULL(boots)
+	for(var/obj/item/mod/module/module in modules)
+		module.mod = null
+		QDEL_NULL(module)
+	..()
+
+/obj/item/mod/control/process(delta_time)
+	if(seconds_electrified > MACHINE_NOT_ELECTRIFIED)
+		seconds_electrified--
+	if(!cell?.charge && active && !activating)
+		power_off()
+		return PROCESS_KILL
+	var/malfunctioning_charge_drain = 0
+	if(malfunctioning)
+		malfunctioning_charge_drain = rand(1,20)
+	cell.charge = max(0, cell.charge - (cell_drain + malfunctioning_charge_drain)*delta_time)
+	for(var/obj/item/mod/module/module in modules)
+		if(malfunctioning && module.active && DT_PROB(5, delta_time))
+			module.on_deactivation()
+		module.on_process(delta_time)
+
+/obj/item/mod/control/equipped(mob/user, slot)
+	..()
+	if(slot == ITEM_SLOT_BACK)
+		wearer = user
+		RegisterSignal(wearer, COMSIG_ATOM_EXITED, .proc/on_exit)
+	else if(wearer)
+		UnregisterSignal(wearer, COMSIG_ATOM_EXITED)
+		wearer = null
+
+/obj/item/mod/control/dropped(mob/user)
+	..()
+	wearer = null
+
+/obj/item/mod/control/item_action_slot_check(slot)
+	if(slot == ITEM_SLOT_BACK)
+		return TRUE
+
+/obj/item/mod/control/allow_attack_hand_drop(mob/user)
+	var/mob/living/carbon/carbon_user = user
+	if(!istype(carbon_user) || src != carbon_user.back)
+		return ..()
+	for(var/obj/item/part in mod_parts)
+		if(part.loc != src)
+			to_chat(carbon_user, "<span class='warning'>ERROR: At least one of the parts are still on your body, please retract them and try again.</span>")
+			playsound(src, 'sound/machines/scanbuzz.ogg', 25, FALSE)
+			return FALSE
+
+/obj/item/mod/control/MouseDrop(atom/over_object)
+	if(src != wearer?.back || !istype(over_object, /atom/movable/screen/inventory/hand))
+		return ..()
+	for(var/obj/item/part in mod_parts)
+		if(part.loc != src)
+			to_chat(wearer, "<span class='warning'>ERROR: At least one of the parts are still on your body, please retract them and try again.</span>")
+			playsound(src, 'sound/machines/scanbuzz.ogg', 25, FALSE)
+			return
+	if(!wearer.incapacitated())
+		var/atom/movable/screen/inventory/hand/ui_hand = over_object
+		if(wearer.putItemFromInventoryInHandIfPossible(src, ui_hand.held_index))
+			add_fingerprint(usr)
+			return ..()
+
+/obj/item/mod/control/attack_hand(mob/user)
+	if(seconds_electrified && cell.charge)
+		if(shock(user))
+			return
+	if(open && cell && loc == user)
+		to_chat(user, "<span class='notice'>You start removing [cell].</span>")
+		if(do_after(user, 50, target = src))
+			to_chat(user, "<span class='notice'>You remove [cell].</span>")
+			playsound(src, 'sound/machines/click.ogg', 50, TRUE)
+			if(!user.put_in_hands(cell))
+				cell.forceMove(drop_location())
+		return
+	return ..()
+
+/obj/item/mod/control/screwdriver_act(mob/living/user, obj/item/I)
+	if(..())
+		return TRUE
+	if(active || activating)
```

Comment:
what if you initiate the screwdriver_act() do_after(), THEN start the activation sequence of your suit during that do_after()?